src/DotVVM.Framework/DependencyInjection/DotVVMServiceCollectionExtensions.cs

```diff
@@ -29,12 +29,11 @@ public static class DotvvmServiceCollectionExtensions
         /// Adds essential DotVVM services to the specified <see cref="IServiceCollection" />.
         /// </summary>
         /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
-        /// <param name="allowDebugServices">If the vs-diagnostics services should be registered</param>
-        public static IServiceCollection RegisterDotVVMServices(IServiceCollection services, bool allowDebugServices = true)
+        public static IServiceCollection RegisterDotVVMServices(IServiceCollection services)
         {
             services.AddOptions();
 
-            if (allowDebugServices) services.AddDiagnosticServices();
+            services.AddDiagnosticServices();
```

Comment:
Do we really want to register these services on production?

Reply:
They already was registered on production. `bool allowDebugServices = true` was never false on production. I checked that even when they are registered they are not used when Debug is true.

Reply:
That seems like a bug.

Don't tell me that DiagnosticsRenderer is not used when it's registered like that: https://github.com/riganti/dotvvm/blob/master/src/DotVVM.Framework/DependencyInjection/DotVVMServiceCollectionExtensions.cs#L99

Reply:
Inside of these services is condition that if config.Debug is false then diagnostic sevices are not used.